apps/frontend/src/components/fap/Proposals/AssignProposalsToFaps.tsx

```diff
@@ -0,0 +1,161 @@
+import Alert from '@mui/material/Alert';
+import Button from '@mui/material/Button';
+import Container from '@mui/material/Container';
+import Grid from '@mui/material/Grid';
+import InputAdornment from '@mui/material/InputAdornment';
+import Typography from '@mui/material/Typography';
+import { Form, Formik } from 'formik';
+import React, { useContext } from 'react';
+
+import FormikUIAutocomplete from 'components/common/FormikUIAutocomplete';
+import { UserContext } from 'context/UserContextProvider';
+import {
+  FapInstrument,
+  FapInstrumentInput,
+  ProposalViewInstrument,
+  UserRole,
+} from 'generated/sdk';
+import { useFapsData } from 'hooks/fap/useFapsData';
+import { getUniqueArrayBy } from 'utils/helperFunctions';
+
+type AssignProposalToFapsProps = {
+  close: () => void;
+  assignProposalsToFaps: (
+    fapInstruments: FapInstrumentInput[]
+  ) => Promise<void>;
+  proposalInstruments: (ProposalViewInstrument[] | null)[];
+  proposalFapInstruments?: (FapInstrument | null)[];
+};
+
+const AssignProposalsToFaps = ({
+  close,
+  assignProposalsToFaps,
+  proposalInstruments,
+  proposalFapInstruments,
+}: AssignProposalToFapsProps) => {
+  const { currentRole } = useContext(UserContext);
+  const { faps, loadingFaps } = useFapsData({
+    filter: '',
+    active: true,
+    role: currentRole as UserRole,
+  });
+
+  const proposalsUniqueInstruments: ProposalViewInstrument[] = getUniqueArrayBy(
+    proposalInstruments.flat(),
+    'id'
+  );
+
+  const allSelectedProposalsHaveSameInstruments = [
+    ...proposalInstruments,
+  ].every(
```

Comment:
Why is this spreading an array into a new array? Is there a reason we can't just call `proposalInstruments.every()` directly?

Reply:
This is because the `sort` function inside the `every` mutates the original array and I don't want to do that. It is better to work with the array copy instead of modifying the original array.

Reply:
But to be more clear I will define it on the top with:
const proposalInstrumentsCopy = [...proposalInstruments];
and then after that use the copy array directly.